swri_roscpp/include/swri_roscpp/dynamic_parameters.h

```diff
@@ -0,0 +1,450 @@
+#ifndef SWRI_ROSCPP_DYNAMIC_PARAMETERS_H_
+#define SWRI_ROSCPP_DYNAMIC_PARAMETERS_H_
+
+#include <map>
+#include <string>
+
+#include <boost/thread/mutex.hpp>
+
+#include <ros/console.h>
+#include <ros/node_handle.h>
+
+#include <dynamic_reconfigure/Config.h>
+#include <dynamic_reconfigure/ConfigDescription.h>
+#include <dynamic_reconfigure/GroupState.h>
+#include <dynamic_reconfigure/Reconfigure.h>
+
+namespace swri
+{
+  struct DynamicValue
+  {
+    enum Type
+    {
+      Bool = 0,
+      Float = 1,
+      Double = 2,
+      Int = 3,
+      String = 4
+    };
+
+    Type type;
+    std::string name;
+    std::string description;
+
+    void* pointer;//pointer to the parameter to update on change
+
+    // Defaults, maximum and minimum values for this parameter
+    union
+    {
+      double d;
+      bool b;
+      int i;
+    } Default;
+    union
+    {
+      double d;
+      int i;
+    } Min;
+    union
+    {
+      double d;
+      int i;
+    } Max;
+    std::string default_string;// to get around union issues with strings
+  };
+
+  class DynamicParameters
+  {
+    ros::Publisher descr_pub_;
+    ros::Publisher update_pub_;
+    ros::ServiceServer set_service_;
+    ros::NodeHandle nh_;
+
+    std::map<std::string, DynamicValue> values_;
+
+    boost::mutex mutex_;
+
+    bool setConfigCallback(dynamic_reconfigure::Reconfigure::Request &req,
+          dynamic_reconfigure::Reconfigure::Response &rsp)
+    {
+      ROS_INFO("Got configuration change request");
```

Comment:
I'm not sure whether I'd rather have this log level be `INFO` or `TRACE`.  Would everybody who has reconfigurable parameters want this to be printed every time something is changed?

Reply:
Scratch that, after taking a break for a few weeks I forgot that ROS didn't have a `TRACE` level.  I guess `DEBUG` works, too.